haskell/src/Util.hs

```diff
@@ -4,6 +4,7 @@
 -- | Read a contract at compile time.
 module Util
   ( fetchContract
```

Comment:
:thinking: in newer morley versions there are utilities that could make us avoid these utility definitions, possibly it's worth updating sooner rather than later.

Since this PR is almost ready however, so I suppose it's better to do so after this is merged.

Reply:
Yeah, I've updated this as part of this PR (by using `embedContractM` instead): https://github.com/serokell/segmented-cfmm/pull/18#discussion_r700461264

Let me know if this is good enough.

Reply:
That's OK, since that PR is based on this one I'll keep any comment related to updating deps for it :ok_hand: